q2cli/tests/test_tools.py

```diff
@@ -130,6 +135,70 @@ def test_merged_metadata(self):
         self.assertIn("IDS:  1", result.output)  # only 1 ID is shared
         self.assertIn("COLUMNS:  4", result.output)
 
+    def test_export_to_dir_w_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequenceDirectoryFormat'
+        ])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertTrue(os.path.isdir(output_path))
+
+    def test_export_to_dir_no_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path
+        ])
+
+        self.assertTrue(os.path.isdir(output_path))
+        self.assertIn('index.html', os.listdir(output_path))
+        self.assertIn('index.tsv', os.listdir(output_path))
+
+    def test_export_to_file(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequenceFormatV2'
+            ])
+
+        file = open(output_path, 'r').read()
+        self.assertEqual(result.exit_code, 0)
+        self.assertIn('0', file)
+        self.assertIn('42', file)
+        self.assertIn('43', file)
+
+    def test_export_visualization_to_dir(self):
+        output_path = os.path.join(self.tempdir, 'output/viz.qza')
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path
+        ])
+
+        self.assertIn('index.html', os.listdir(output_path))
+        self.assertIn('index.tsv', os.listdir(output_path))
+        self.assertTrue(os.path.isdir(output_path))
+
+    def test_export_visualization_w_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path,
+            '--output-format', 'IntSequenceDirectoryFormat'
+        ])
+
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('visualization', result.output)
+        self.assertIn('--output-format', result.output)
+
+    def test_export_path_file_is_replaced(self):
+        output_path = os.path.join(self.tempdir, 'helloWorld')
+        os.mkdir(output_path)
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+        ])
+
+        self.assertNotIn('helloWorld', os.listdir(output_path))
```

Comment:
This isn't the right kind of assertion, remember, we're looking to see that the file is replaced.
The rough idea here is right, but we need to look inside a file not a directory.

Reply:
Yeah, you're right! Thanks for the reminding me.